ts/api/api_test.ts

```diff
@@ -288,4 +258,173 @@ describe('OpenYolo API', () => {
       });
     });
   });
+
+  describe('OpenYoloApiImpl', () => {
+    const wrapBrowserError =
+        OpenYoloInternalError.browserWrappingRequired().toExposedError();
+    const otherError =
+        OpenYoloInternalError.noCredentialsAvailable().toExposedError();
+
+    let openYoloApiImpl: OpenYoloApiImpl;
+    let timeoutRacerSpy: jasmine.SpyObj<TimeoutRacer>;
+
+    beforeEach(() => {
+      const frameManager =
+          jasmine.createSpyObj('ProviderFrameElement', ['display']);
+      openYoloApiImpl = new OpenYoloApiImpl(frameManager, secureChannelSpy);
+      timeoutRacerSpy = jasmine.createSpyObj('TimeoutRacer', ['race']);
+    });
+
+    type methodSignatures =
+        {[key in keyof OpenYoloApi]: typeof OpenYoloApiImpl.prototype[key]};
+
+    /**
+     * Tests the behavior of operations implementations.
+     * @param methodName The method of OpenYoloApi being tested.
+     * @param operationRequest The request class's prototype being mocked.
+     * @param options The request's options.
+     * @param expectedResult The expected result when successful.
+     */
+    function testOperationImpl<M extends keyof methodSignatures, Opt, Res>(
+        methodName: M,
+        operationRequest: RelayRequest<Res, Opt>,
+        options: Opt,
+        expectedResult: Res) {
+      let dispatchSpy: jasmine.Spy;
+      let navCredentialsSpy: jasmine.Spy;
+
+      beforeEach(() => {
+        dispatchSpy = spyOn(operationRequest, 'dispatch');
+        navCredentialsSpy = spyOn(NavigatorCredentials.prototype, methodName);
+      });
+
+      it('dispatches the request', (done) => {
+        dispatchSpy.and.returnValue(Promise.resolve(expectedResult));
+        (openYoloApiImpl[methodName] as methodSignatures[M])(
```

Comment:
I think it complained about untyped method, as the argument `methodName` would not give typing information on the method called, but the template would.